tests/end_to_end/utils/summary_helper.py

```diff
@@ -41,13 +44,15 @@ def get_aggregated_accuracy(agg_log_file):
         0.15911591053009033, 'round': 0}
     """
     try:
-        with open(agg_log_file, 'r') as f:
-            for line in f:
-                if "'metric_origin': 'aggregator'" in line and "aggregated_model_validation" in line:
-                    line = line.split("aggregator.py:")[0].strip()
-                    # If the line does not contain closing bracket "}", then concatenate the next line
-                    reqd_line = line if "}" in line else line + next(f).strip()
-                    agg_accuracy = eval(reqd_line.split("METRIC")[1].strip('"'))["metric_value"]
+        with open(agg_log_file, "r") as file:
+            lines = file.readlines()
+            for line in lines:
+                match = re.search(
+                    r"'metric_origin': 'aggregator', 'task_name': 'aggregated_model_validation', 'metric_name': 'accuracy', 'metric_value':\s*([\d.]+)",
+                    line,
+                )
```

Comment:
Regex searching for metrics is not a good approach. This will break immediately once logging style changes (FYI - that may happen soon). Better way is to write metrics to a CSV/JSON and read structured data as-is.

@payalcha can provide context.

Reply:
I discussed it with Payal. If the callback is not implemented yet, we will need to change logic in `aggregator.py` to redirect the metrics to a file. 
If that is fine, I can modify the file, else will wait for your PR to get merged.

Reply:
We discussed about this change. I will wait for callback implementation for metrics, and then use it (instead of current method of parsing through the file)